include/common/utils/concepts.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <concepts>
+#include <fstream>
+#include <functional>
+#include <type_traits>
+
+namespace utils
+{
+
+template <typename T>
+concept Enumeration = std::is_enum_v<T>;
+
+// clang-format off
+
+template <typename T, typename D>
+concept CoercibleTo = requires ( T source ) {
```

Comment:
I could not come up with another name, because this is not ```std::convertible_to```, because a type can be coerced with an explicit cast, but the standard ```std::convertible_to``` requires the type to be implicitly convertible in both ways.
Maybe this can be called ```ExplicitlyConvertibleTo```